packages/scaffold-ui/src/views/w3m-select-addresses-view/index.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  AccountController,
+  ConnectionController,
+  ModalController,
+  OptionsController,
+  RouterController,
+  type AccountType
+} from '@web3modal/core'
+import { customElement } from '@web3modal/ui'
+import { LitElement, html } from 'lit'
+import { state } from 'lit/decorators.js'
+import { ifDefined } from 'lit/directives/if-defined.js'
+import styles from './styles.js'
+
+@customElement('w3m-select-addresses-view')
+export class W3mSelectAddressesView extends LitElement {
+  public static override styles = styles
+  // -- Members ------------------------------------------- //
+  private readonly metadata = OptionsController.state.metadata
+  public allAccounts: AccountType[] = AccountController.state.allAccounts
+  private selectedAccounts: AccountType[] = AccountController.state.allAccounts
+  private selectAll = true
+  private approved = false
+
+  @state() private isApproving = false
+  constructor() {
+    super()
+    AccountController.subscribeKey('allAccounts', allAccounts => {
+      this.allAccounts = allAccounts
+      this.requestUpdate()
```

Comment:
is this forced update required? have you tried making `allAccounts` a `@state` variable so that changes trigger a re-remder?